editors/code/src/Commands/compileSM.ts

```diff
@@ -23,90 +23,65 @@ let output: OutputChannel;
  * @returns Promise
  */
 export async function run(args: URI): Promise<void> {
-  const uri = args === undefined ? window.activeTextEditor.document.uri : args;
-  const workspaceFolder = Workspace.getWorkspaceFolder(uri);
-
-  const alwaysCompileMainPath: boolean = Workspace.getConfiguration(
-    "sourcepawn",
-    workspaceFolder
-  ).get<boolean>("MainPathCompilation");
-
-  // Decide which file to compile here.
   let fileToCompilePath: string;
-  if (alwaysCompileMainPath) {
-    const params: ProjectMainPathParams = { uri: uri.toString() };
-    const mainUri = await getCtxFromUri(uri)?.client.sendRequest(
-      projectMainPath,
-      params
-    );
-    if (mainUri === undefined) {
-      fileToCompilePath = uri.fsPath;
-    } else {
-      fileToCompilePath = URI.parse(mainUri).fsPath;
+
+  // If we always compile the main path, we always ignore the path of the current editor
+  const workspaceFolder = workspace.getWorkspaceFolder(lastActiveEditor.document.uri);
+  const compileMainPath: boolean = getConfig(Section.SourcePawn, "MainPathCompilation", workspaceFolder);
+  if (compileMainPath) {
+    fileToCompilePath = await getMainCompilationFile()
+  }
+  // Else, we take the arguments, or we take the last active editor's path
+  else {
+    if (args) {
+      fileToCompilePath = args.fsPath;
+    }
+    else {
+      fileToCompilePath = lastActiveEditor.document.uri.fsPath;
     }
-  } else {
-    fileToCompilePath = uri.fsPath;
   }
 
-  const scriptingFolderPath = dirname(fileToCompilePath);
-
   // Don't compile if it's not a .sp file.
-  if (extname(fileToCompilePath) !== ".sp") {
-    window.showErrorMessage("Not a .sp file, aborting");
+  if (!isSPFile(fileToCompilePath)) {
+    window.showErrorMessage("The selected file is not a SourcePawn script.");
```

Comment:
```suggestion
    window.showErrorMessage("The current file is not a SourcePawn script.");
```

Running the command with Ctrl+shift+D does not really "select" the file.